src/puppetlabs/puppetdb/cli/services.clj

```diff
@@ -249,9 +249,13 @@
 (defn maybe-check-for-updates
   [config read-db]
   (if (conf/foss? config)
-    (-> config
-        conf/update-server
-        (version/check-for-updates! read-db))
+    (let [job-pool (mk-pool)
```

Comment:
Ah yeah, do you think it'll be smarter to move this check to use the same job pool, then? Basically I'm thinking move the `maybe-check-for-updates` conditional into where the `let` is just below and just pass the job-pool that was created there into the function. I could assoc this job pool onto context to be shut down later, but this job is pretty small and could probably use the same one.